unbalanced_dataset/under_sampling/instance_hardness_threshold.py

```diff
@@ -0,0 +1,240 @@
+"""Class to perform under-sampling based on the instance hardness
+threshold."""
+from __future__ import print_function
+from __future__ import division
+
+import numpy as np
+
+from collections import Counter
+
+from scipy.stats import mode
+
+from sklearn.utils import check_X_y
+from sklearn.neighbors import NearestNeighbors
+from sklearn.cross_validation import StratifiedKFold
+
+from .under_sampler import UnderSampler
+
+
+class InstanceHardnessThreshold(UnderSampler):
+    """Class to perform under-sampling based on the instance hardness 
+    threshold.
+
+    Parameters
+    ----------
+    estimator : sklearn classifier
+        Classifier to be used in to estimate instance hardness of the samples.
+
+    threshold : float, optional (default=0.3)
+        Threshold to be used when excluding samples (0.01 to 0.99).
+
+    mode: str, optional (default='maj')
+        - If 'maj', only samples of the majority class are excluded.
+        - If 'all', samples of all classes are excluded.
+
+    cv : int, optional (default=5)
+        Number of folds to be used when estimating samples' instance hardness.
+
+    return_indices : bool, optional (default=False)
+        Either to return or not the indices which will be selected from
+        the majority class.
+
+    random_state : int or None, optional (default=None)
+        Seed for random number generation.
+
+    verbose : bool, optional (default=True)
+        Boolean to either or not print information about the processing
+
+    n_jobs : int, optional (default=-1)
+        The number of thread to open when it is possible.
+
```

Comment:
you have two empty lines (50, 51). If I remember you should not have double space lines, usually.